src/common/src/system_param/mod.rs

```diff
@@ -83,23 +110,42 @@ macro_rules! def_key {
 
 for_all_params!(def_key);
 
-/// Define default value functions.
-macro_rules! def_default {
-    ($({ $field:ident, $type:ty, $default:expr, $is_mutable:expr },)*) => {
-        pub mod default {
-            $(
-                pub fn $field() -> Option<$type> {
+/// Define default value functions returning `Option`.
+macro_rules! def_default_opt {
+    ($({ $field:ident, $type:ty, $default: expr, $($rest:tt)* },)*) => {
+        $(
+            paste::paste!(
+                pub fn [<$field _opt>]() -> Option<$type> {
                     $default
                 }
-            )*
+            );
+        )*
+    };
+}
+
+/// Define default value functions for those with `Some` default values.
+macro_rules! def_default {
+    ($({ $field:ident, $type:ty, $default: expr, $($rest:tt)* },)*) => {
+        $(
+            def_default!(@ $field, $type, $default);
+        )*
+    };
+    (@ $field:ident, $type:ty, None) => {};
+    (@ $field:ident, $type:ty, $default: expr) => {
```

Comment:
![image.png](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/JgO8A8bpio3B4YlQByok/00cee040-a074-4196-9b46-37cc4d9a29dc.png)

Reply:
Where is this non-opt version used?

Reply:
In the Reader Impl